inst/assumptions.m

```diff
@@ -111,10 +111,9 @@
       'd = x._assumptions.generator'
       'if d == {}:'
       '    astr = ""'
-      'elif all(d.values()):'  % all True so list them
-      '    astr = ", ".join(sorted([str(i) for i in d.keys()]))'
-      'else:'  % more complicated case, just the raw dict
-      '    astr = str(d)'
+      'else:'  % all True so list them
+      '    val, key = list(d.values()), list(d.keys())' % In Python3 d.values() don't support indexing
+      '    astr = ", ".join(sorted([(("" if val[i] else "~") + str(key[i])) for i in range(len(d.keys()))]))'
```

Comment:
I forget if Sympy assumptions have to be either true or false... Can they be None?

I think this code assumes each value of the dict is either true or false.